pkg/controller/jobpost/jobpost.go

```diff
@@ -0,0 +1,89 @@
+package jobpost
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-playground/validator/v10"
+
+	"github.com/google/uuid"
+	"github.com/hngprojects/hng_boilerplate_golang_web/external/request"
+	"github.com/hngprojects/hng_boilerplate_golang_web/internal/models"
+	"github.com/hngprojects/hng_boilerplate_golang_web/pkg/repository/storage"
+	service "github.com/hngprojects/hng_boilerplate_golang_web/services/jobpost"
+	"github.com/hngprojects/hng_boilerplate_golang_web/utility"
+)
+
+type Controller struct {
+  Db        *storage.Database
+  Validator *validator.Validate
+  Logger    *utility.Logger
+  ExtReq    request.ExternalRequest 
+}
+
+func (base *Controller) CreateJobPost(c *gin.Context) {
+	var req models.JobPost
```

Comment:
create a request model for this and specify all required fields, check the other models for reference

Reply:
still waiting for this to be done